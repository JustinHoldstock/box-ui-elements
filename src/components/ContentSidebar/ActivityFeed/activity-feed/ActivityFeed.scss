@import '../../../styles/variables';

.box-ui-activity-feed {
    display: flex;
    flex-direction: column;
    height: 100%;
    position: relative;

    .box-ui-activity-feed-items-container {
        align-items: center;
        display: flex;
        flex: 1 1;
        flex-direction: column;
        overflow-y: auto;
        overflow-x: hidden;
        position: relative;

        .box-ui-version-item {
            &:first-child .box-ui-version {
                margin-top: 0;
            }
        }

        .box-ui-activity-feed-task,
        .box-ui-activity-feed-comment {
            border-bottom: 1px solid $ffive;
            padding-bottom: 8px;

            &:first-child .box-ui-comment {
                margin-top: 0;
            }
        }

        .box-ui-activity-feed-task {
            padding-bottom: 28px;
        }
    }

    .box-ui-activity-feed-active-state {
        bottom: 0;
        left: 0;
        margin: 9px 20px;
        position: absolute;
        right: 0;
        top: 0;

        > li:last-child {
            margin-bottom: 10px;
        }
    }

    .box-ui-activity-feed-empty-state {
        align-items: center;
        display: flex;
        flex-direction: column;

        .empty-state-cta {
            align-items: center;
            display: flex;
            flex-direction: column;

            aside {
                margin-top: 6px;
                text-align: center;
                width: 200px;
            }
        }

        .box-ui-activity-feed-empty-state-illustration {
            margin-bottom: 15px;
            margin-top: 60px;
        }
    }

    .box-ui-activity-feed-comment-input {
        background-color: $white;
        margin-bottom: 4px;
        max-height: 500px;
        min-height: 93px;

        &.is-disabled {
            opacity: .4;
            pointer-events: none;
        }

        .selector-dropdown-wrapper {
            width: 100%;
        }
    }
}
