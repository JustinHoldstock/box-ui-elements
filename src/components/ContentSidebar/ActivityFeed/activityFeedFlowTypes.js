/**
 * @flow
 * @file Flow types for Activity Feed
 * @author Box
 */

export type Comments = {
    create: Function,
    delete: Function
};

export type Tasks = {
    create?: Function,
    delete?: Function,
    edit?: Function,
    onTaskAssignmentUpdate?: Function
};

export type Contacts = {
    getApproverWithQuery: Function,
    getMentionWithQuery: Function
};

export type Versions = {
    info: Function
};

export type SelectorItem = {
    id?: string,
    name: string
};

export type SelectorItems = Array<SelectorItem>;

export type InputState = {
    approverSelectorContacts?: SelectorItems,
    mentionSelectorContacts?: SelectorItems,
    currentUser: User,
    isDisabled?: boolean
};

export type Item = {
    type?: 'comment' | 'task' | 'file_version' | 'keywords',
    createdAt?: any,
    createdBy: User,
    id: string
};

export type Translations = {
    translationEnabled?: boolean,
    onTranslate?: boolean
};
