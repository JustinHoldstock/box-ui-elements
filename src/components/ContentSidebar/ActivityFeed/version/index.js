export CollapsedVersion from './CollapsedVersion';
export VersionError from './VersionError';
export default from './Version';
