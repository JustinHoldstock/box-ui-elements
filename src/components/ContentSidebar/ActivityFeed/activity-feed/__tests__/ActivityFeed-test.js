import React from 'react';
import { mount, shallow } from 'enzyme';

import ActivityFeed from '../ActivityFeed';

const feedState = [
    {
        type: 'comment',
        id: '123',
        createdAt: Date.now(),
        taggedMessage: 'test @[123:Jeezy] @[10:Kanye West]',
        createdBy: { name: 'Akon', id: 11 }
    },
    {
        type: 'task',
        id: '1234',
        createdAt: Date.now(),
        taggedMessage: 'test',
        createdBy: { name: 'JayZ', id: 10 },
        assignees: []
    }
];
const currentUser = { name: 'Kanye West', id: 10 };

const allHandlers = {
    comments: {
        create: jest.fn()
    },
    tasks: {
        create: jest.fn()
    },
    contacts: {
        getApproverWithQuery: jest.fn(),
        getMentionWithQuery: jest.fn()
    }
};

describe('features/activity-feed/activity-feed/ActivityFeed', () => {
    test('should correctly render empty state', () => {
        const wrapper = shallow(<ActivityFeed inputState={{ currentUser }} />);

        expect(wrapper.find('EmptyState').length).toEqual(1);
        expect(wrapper.find('ApprovalCommentForm').length).toEqual(0);
    });

    test('should correctly render empty state with loading indicator', () => {
        const wrapper = shallow(<ActivityFeed inputState={{ currentUser }} isLoading />);

        expect(wrapper.find('EmptyState').length).toEqual(1);
        expect(wrapper.find('LoadingIndicator').length).toEqual(0);
    });

    test('should render approval comment form if comment submit handler is passed', () => {
        const wrapper = shallow(<ActivityFeed inputState={{ currentUser }} handlers={allHandlers} />);

        expect(wrapper.find('ApprovalCommentForm').length).toEqual(1);
    });

    test('should correctly render activity state', () => {
        const wrapper = shallow(<ActivityFeed inputState={{ currentUser }} feedState={feedState} />);

        expect(wrapper.find('ActiveState').length).toEqual(1);
    });

    test('should not expose add approval ui if task submit handler is not passed', () => {
        const noTaskHandler = {
            ...allHandlers,
            tasks: null
        };
        const wrapper = shallow(<ActivityFeed inputState={{ currentUser }} handlers={noTaskHandler} />);

        expect(wrapper.find('[name="addApproval"]').length).toEqual(0);
    });

    test('should show input when approvalCommentFormFocusHandler is called', () => {
        const wrapper = shallow(<ActivityFeed inputState={{ currentUser }} handlers={allHandlers} />);

        const instance = wrapper.instance();
        instance.approvalCommentFormFocusHandler();

        expect(wrapper.state('isInputOpen')).toBe(true);
    });

    test('should hide input when approvalCommentFormCancelHandler is called', () => {
        const wrapper = shallow(<ActivityFeed inputState={{ currentUser }} handlers={allHandlers} />);

        const instance = wrapper.instance();
        instance.approvalCommentFormFocusHandler();
        expect(wrapper.state('isInputOpen')).toBe(true);

        instance.approvalCommentFormCancelHandler();
        expect(wrapper.state('isInputOpen')).toBe(false);
    });

    test('should call create comment handler and close input on valid comment submit', () => {
        const createCommentSpy = jest.fn();
        allHandlers.comments.create = createCommentSpy;
        const wrapper = mount(<ActivityFeed inputState={{ currentUser }} handlers={allHandlers} />);

        const instance = wrapper.instance();
        const approvalCommentForm = wrapper.find('ApprovalCommentForm').at(0);

        instance.approvalCommentFormFocusHandler();
        expect(wrapper.state('isInputOpen')).toBe(true);

        approvalCommentForm.prop('createComment')();
        expect(wrapper.state('isInputOpen')).toBe(false);
        expect(createCommentSpy.mock.calls).toEqual(1);
    });

    test('should call create task handler and close input on valid task submit', () => {
        const createTaskSpy = jest.fn();
        allHandlers.tasks.create = createTaskSpy;
        const wrapper = mount(<ActivityFeed inputState={{ currentUser }} handlers={allHandlers} />);

        const instance = wrapper.instance();
        const approvalCommentForm = wrapper.find('ApprovalCommentForm').at(0);

        instance.approvalCommentFormFocusHandler();
        expect(wrapper.state('isInputOpen')).toBe(true);

        approvalCommentForm.prop('createTask')();
        expect(wrapper.state('isInputOpen')).toBe(false);
        expect(createTaskSpy.mock.calls).toEqual(1);
    });

    test('should stop event propagation onKeyDown', () => {
        const wrapper = shallow(<ActivityFeed inputState={{ currentUser }} handlers={allHandlers} />);
        const stopPropagationSpy = jest.fn();
        wrapper.find('.box-ui-activity-feed').simulate('keydown', {
            nativeEvent: {
                stopImmediatePropagation: stopPropagationSpy
            }
        });
        expect(stopPropagationSpy).toHaveBeenCalled();
    });
});
