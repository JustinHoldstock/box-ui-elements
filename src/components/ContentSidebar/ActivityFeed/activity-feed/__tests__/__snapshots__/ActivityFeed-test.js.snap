// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/activity-feed/activity-feed/ActivityFeed should correctly render activity state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivityFeed
    feedState={
      Array [
        Object {
          "createdAt": 1234567890,
          "createdBy": Object {
            "id": 11,
            "name": "Akon",
          },
          "id": "123",
          "taggedMessage": "test @[123:Jeezy] @[10:Kanye West]",
          "type": "comment",
        },
        Object {
          "assignees": Array [],
          "createdAt": 1234567891,
          "createdBy": Object {
            "id": 10,
            "name": "JayZ",
          },
          "id": "1234",
          "taggedMessage": "test",
          "type": "task",
        },
      ]
    }
    inputState={
      Object {
        "currentUser": Object {
          "id": 10,
          "name": "Kanye West",
        },
      }
    }
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="box-ui-activity-feed-items-container"
        >
          <ActiveState
            currentUser={
              Object {
                "id": 10,
                "name": "Kanye West",
              }
            }
            handlers={undefined}
            inputState={
              Object {
                "currentUser": Object {
                  "id": 10,
                  "name": "Kanye West",
                },
              }
            }
            items={
              Array [
                Object {
                  "createdAt": 1234567890,
                  "createdBy": Object {
                    "id": 11,
                    "name": "Akon",
                  },
                  "id": "123",
                  "taggedMessage": "test @[123:Jeezy] @[10:Kanye West]",
                  "type": "comment",
                },
                Object {
                  "assignees": Array [],
                  "createdAt": 1234567891,
                  "createdBy": Object {
                    "id": 10,
                    "name": "JayZ",
                  },
                  "id": "1234",
                  "taggedMessage": "test",
                  "type": "task",
                },
              ]
            }
            onCommentDelete={undefined}
            onTaskAssignmentUpdate={undefined}
            onTaskDelete={undefined}
            onTaskEdit={undefined}
            onVersionInfo={undefined}
            translations={undefined}
          />
        </div>,
        null,
      ],
      "className": "box-ui-activity-feed",
      "onKeyDown": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ActiveState
            currentUser={
              Object {
                "id": 10,
                "name": "Kanye West",
              }
            }
            handlers={undefined}
            inputState={
              Object {
                "currentUser": Object {
                  "id": 10,
                  "name": "Kanye West",
                },
              }
            }
            items={
              Array [
                Object {
                  "createdAt": 1234567890,
                  "createdBy": Object {
                    "id": 11,
                    "name": "Akon",
                  },
                  "id": "123",
                  "taggedMessage": "test @[123:Jeezy] @[10:Kanye West]",
                  "type": "comment",
                },
                Object {
                  "assignees": Array [],
                  "createdAt": 1234567891,
                  "createdBy": Object {
                    "id": 10,
                    "name": "JayZ",
                  },
                  "id": "1234",
                  "taggedMessage": "test",
                  "type": "task",
                },
              ]
            }
            onCommentDelete={undefined}
            onTaskAssignmentUpdate={undefined}
            onTaskDelete={undefined}
            onTaskEdit={undefined}
            onVersionInfo={undefined}
            translations={undefined}
          />,
          "className": "box-ui-activity-feed-items-container",
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentUser": Object {
              "id": 10,
              "name": "Kanye West",
            },
            "handlers": undefined,
            "inputState": Object {
              "currentUser": Object {
                "id": 10,
                "name": "Kanye West",
              },
            },
            "items": Array [
              Object {
                "createdAt": 1234567890,
                "createdBy": Object {
                  "id": 11,
                  "name": "Akon",
                },
                "id": "123",
                "taggedMessage": "test @[123:Jeezy] @[10:Kanye West]",
                "type": "comment",
              },
              Object {
                "assignees": Array [],
                "createdAt": 1234567891,
                "createdBy": Object {
                  "id": 10,
                  "name": "JayZ",
                },
                "id": "1234",
                "taggedMessage": "test",
                "type": "task",
              },
            ],
            "onCommentDelete": undefined,
            "onTaskAssignmentUpdate": undefined,
            "onTaskDelete": undefined,
            "onTaskEdit": undefined,
            "onVersionInfo": undefined,
            "translations": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="box-ui-activity-feed-items-container"
          >
            <ActiveState
              currentUser={
                Object {
                  "id": 10,
                  "name": "Kanye West",
                }
              }
              handlers={undefined}
              inputState={
                Object {
                  "currentUser": Object {
                    "id": 10,
                    "name": "Kanye West",
                  },
                }
              }
              items={
                Array [
                  Object {
                    "createdAt": 1234567890,
                    "createdBy": Object {
                      "id": 11,
                      "name": "Akon",
                    },
                    "id": "123",
                    "taggedMessage": "test @[123:Jeezy] @[10:Kanye West]",
                    "type": "comment",
                  },
                  Object {
                    "assignees": Array [],
                    "createdAt": 1234567891,
                    "createdBy": Object {
                      "id": 10,
                      "name": "JayZ",
                    },
                    "id": "1234",
                    "taggedMessage": "test",
                    "type": "task",
                  },
                ]
              }
              onCommentDelete={undefined}
              onTaskAssignmentUpdate={undefined}
              onTaskDelete={undefined}
              onTaskEdit={undefined}
              onVersionInfo={undefined}
              translations={undefined}
            />
          </div>,
          null,
        ],
        "className": "box-ui-activity-feed",
        "onKeyDown": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ActiveState
              currentUser={
                Object {
                  "id": 10,
                  "name": "Kanye West",
                }
              }
              handlers={undefined}
              inputState={
                Object {
                  "currentUser": Object {
                    "id": 10,
                    "name": "Kanye West",
                  },
                }
              }
              items={
                Array [
                  Object {
                    "createdAt": 1234567890,
                    "createdBy": Object {
                      "id": 11,
                      "name": "Akon",
                    },
                    "id": "123",
                    "taggedMessage": "test @[123:Jeezy] @[10:Kanye West]",
                    "type": "comment",
                  },
                  Object {
                    "assignees": Array [],
                    "createdAt": 1234567891,
                    "createdBy": Object {
                      "id": 10,
                      "name": "JayZ",
                    },
                    "id": "1234",
                    "taggedMessage": "test",
                    "type": "task",
                  },
                ]
              }
              onCommentDelete={undefined}
              onTaskAssignmentUpdate={undefined}
              onTaskDelete={undefined}
              onTaskEdit={undefined}
              onVersionInfo={undefined}
              translations={undefined}
            />,
            "className": "box-ui-activity-feed-items-container",
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentUser": Object {
                "id": 10,
                "name": "Kanye West",
              },
              "handlers": undefined,
              "inputState": Object {
                "currentUser": Object {
                  "id": 10,
                  "name": "Kanye West",
                },
              },
              "items": Array [
                Object {
                  "createdAt": 1234567890,
                  "createdBy": Object {
                    "id": 11,
                    "name": "Akon",
                  },
                  "id": "123",
                  "taggedMessage": "test @[123:Jeezy] @[10:Kanye West]",
                  "type": "comment",
                },
                Object {
                  "assignees": Array [],
                  "createdAt": 1234567891,
                  "createdBy": Object {
                    "id": 10,
                    "name": "JayZ",
                  },
                  "id": "1234",
                  "taggedMessage": "test",
                  "type": "task",
                },
              ],
              "onCommentDelete": undefined,
              "onTaskAssignmentUpdate": undefined,
              "onTaskDelete": undefined,
              "onTaskEdit": undefined,
              "onVersionInfo": undefined,
              "translations": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`features/activity-feed/activity-feed/ActivityFeed should correctly render empty state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivityFeed
    feedState={Array []}
    inputState={
      Object {
        "currentUser": Object {
          "id": 10,
          "name": "Kanye West",
        },
      }
    }
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="box-ui-activity-feed-items-container"
        >
          <EmptyState
            isLoading={false}
            showCommentMessage={false}
          />
        </div>,
        null,
      ],
      "className": "box-ui-activity-feed",
      "onKeyDown": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <EmptyState
            isLoading={false}
            showCommentMessage={false}
          />,
          "className": "box-ui-activity-feed-items-container",
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isLoading": false,
            "showCommentMessage": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="box-ui-activity-feed-items-container"
          >
            <EmptyState
              isLoading={false}
              showCommentMessage={false}
            />
          </div>,
          null,
        ],
        "className": "box-ui-activity-feed",
        "onKeyDown": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <EmptyState
              isLoading={false}
              showCommentMessage={false}
            />,
            "className": "box-ui-activity-feed-items-container",
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isLoading": false,
              "showCommentMessage": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`features/activity-feed/activity-feed/ActivityFeed should correctly render empty state with loading indicator 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivityFeed
    feedState={Array []}
    inputState={
      Object {
        "currentUser": Object {
          "id": 10,
          "name": "Kanye West",
        },
      }
    }
    isLoading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="box-ui-activity-feed-items-container"
        >
          <EmptyState
            isLoading={true}
            showCommentMessage={false}
          />
        </div>,
        null,
      ],
      "className": "box-ui-activity-feed",
      "onKeyDown": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <EmptyState
            isLoading={true}
            showCommentMessage={false}
          />,
          "className": "box-ui-activity-feed-items-container",
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isLoading": true,
            "showCommentMessage": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="box-ui-activity-feed-items-container"
          >
            <EmptyState
              isLoading={true}
              showCommentMessage={false}
            />
          </div>,
          null,
        ],
        "className": "box-ui-activity-feed",
        "onKeyDown": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <EmptyState
              isLoading={true}
              showCommentMessage={false}
            />,
            "className": "box-ui-activity-feed-items-container",
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isLoading": true,
              "showCommentMessage": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`features/activity-feed/activity-feed/ActivityFeed should render approval comment form if comment submit handler is passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivityFeed
    feedState={Array []}
    handlers={
      Object {
        "comments": Object {
          "create": [MockFunction],
        },
        "contacts": Object {
          "getApproverWithQuery": [MockFunction],
          "getMentionWithQuery": [MockFunction],
        },
        "tasks": Object {
          "create": [MockFunction],
        },
      }
    }
    inputState={
      Object {
        "currentUser": Object {
          "id": 10,
          "name": "Kanye West",
        },
      }
    }
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="box-ui-activity-feed-items-container"
        >
          <EmptyState
            isLoading={false}
            showCommentMessage={true}
          />
        </div>,
        <ApprovalCommentForm
          approverSelectorContacts={undefined}
          className="box-ui-activity-feed-comment-input "
          createComment={[Function]}
          createTask={[Function]}
          getApproverContactsWithQuery={[MockFunction]}
          getMentionContactsWithQuery={[MockFunction]}
          isDisabled={undefined}
          isOpen={false}
          mentionSelectorContacts={undefined}
          onCancel={[Function]}
          onFocus={[Function]}
          onSubmit={[Function]}
          user={
            Object {
              "id": 10,
              "name": "Kanye West",
            }
          }
        />,
      ],
      "className": "box-ui-activity-feed",
      "onKeyDown": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <EmptyState
            isLoading={false}
            showCommentMessage={true}
          />,
          "className": "box-ui-activity-feed-items-container",
        },
        "ref": [Function],
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isLoading": false,
            "showCommentMessage": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "approverSelectorContacts": undefined,
          "className": "box-ui-activity-feed-comment-input ",
          "createComment": [Function],
          "createTask": [Function],
          "getApproverContactsWithQuery": [MockFunction],
          "getMentionContactsWithQuery": [MockFunction],
          "isDisabled": undefined,
          "isOpen": false,
          "mentionSelectorContacts": undefined,
          "onCancel": [Function],
          "onFocus": [Function],
          "onSubmit": [Function],
          "user": Object {
            "id": 10,
            "name": "Kanye West",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="box-ui-activity-feed-items-container"
          >
            <EmptyState
              isLoading={false}
              showCommentMessage={true}
            />
          </div>,
          <ApprovalCommentForm
            approverSelectorContacts={undefined}
            className="box-ui-activity-feed-comment-input "
            createComment={[Function]}
            createTask={[Function]}
            getApproverContactsWithQuery={[MockFunction]}
            getMentionContactsWithQuery={[MockFunction]}
            isDisabled={undefined}
            isOpen={false}
            mentionSelectorContacts={undefined}
            onCancel={[Function]}
            onFocus={[Function]}
            onSubmit={[Function]}
            user={
              Object {
                "id": 10,
                "name": "Kanye West",
              }
            }
          />,
        ],
        "className": "box-ui-activity-feed",
        "onKeyDown": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <EmptyState
              isLoading={false}
              showCommentMessage={true}
            />,
            "className": "box-ui-activity-feed-items-container",
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isLoading": false,
              "showCommentMessage": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "approverSelectorContacts": undefined,
            "className": "box-ui-activity-feed-comment-input ",
            "createComment": [Function],
            "createTask": [Function],
            "getApproverContactsWithQuery": [MockFunction],
            "getMentionContactsWithQuery": [MockFunction],
            "isDisabled": undefined,
            "isOpen": false,
            "mentionSelectorContacts": undefined,
            "onCancel": [Function],
            "onFocus": [Function],
            "onSubmit": [Function],
            "user": Object {
              "id": 10,
              "name": "Kanye West",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
